src/network/networkd-ndisc.c

```diff
@@ -839,6 +839,11 @@ static int ndisc_router_process_route(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return log_link_error_errno(link, r, "Failed to get gateway address from RA: %m");
 
+        if (in6_addr_compare_func(&link->ipv6ll_address, &gateway) == 0) {
```

Comment:
Maybe, it is better to use `link_has_ipv6_address()`. See `ndisc_router_process_default()`.

Reply:
Yes, this is better.